ftplugin/orgmode/plugins/Date.py

```diff
@@ -131,37 +135,48 @@ def _modify_time(cls, startdate, modifier):
 			newdate = startdate + timedelta(days=diff)
 
 		# check for days modifier with appended d
-		match = re.search(u'\+(\d*)d', modifier)
+		match = re.search(u'^(\+|-)(\d*)d', modifier)
 		if match:
-			days = int(match.groups()[0])
-			newdate = startdate + timedelta(days=days)
+			groups = match.groups()
+			op = groups[0]
+			days = int(groups[1])
+			newdate = cls.ops[op](startdate, timedelta(days=days))
```

Comment:
What do you think of using
~~~python
op, days = match.groups()
newdate = ops[op](startdate, timedelta(days=int(days))
~~~